File: app/src/main/java/antoninovitale/dropcodechallenge/util/DebugLog.java
package antoninovitale.dropcodechallenge.util;

import android.util.Log;

import antoninovitale.dropcodechallenge.BuildConfig;

/**
 * Created by a.vitale on 29/08/2017.
 */
public class DebugLog {

    public static void log(String tag, String message, Throwable e) {
        if (BuildConfig.DEBUG) {
            Log.d(tag, message, e);
        }
    }

    public static void log(String tag, String message) {
        if (BuildConfig.DEBUG) {
            Log.d(tag, message);
        }
    }

}

File: app/src/main/java/antoninovitale/punkapi/app/util/DebugLog.kt
package antoninovitale.punkapi.app.util

import android.util.Log
import antoninovitale.punkapi.app.BuildConfig

/**
 * Created by a.vitale on 29/08/2017.
 */
object DebugLog {
    fun log(tag: String?, message: String?, e: Throwable?) {
        if (BuildConfig.DEBUG) {
            Log.d(tag, message ?: "", e)
        }
    }

    fun log(tag: String?, message: String?) {
        if (BuildConfig.DEBUG) {
            Log.d(tag, message ?: "")
        }
    }
}